CMakeLists.txt

```diff
@@ -1,31 +1,209 @@
+		HOMEPAGE_URL https://github.com/KarypisLab/GKlib
+		LANGUAGES C
+)
 
-include_directories(".")
-if(MSVC)
-  include_directories("win32")
-  file(GLOB win32_sources RELATIVE "win32" "*.c")
-else(MSVC)
-  set(win32_sources, "")
-endif(MSVC)
+# Back-porting to PROJECT_IS_TOP_LEVEL to older cmake
+# TODO: Remove when requiring cmake >= 3.21
+if (NOT DEFINED Spglib_IS_TOP_LEVEL)
+	if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+		set(PROJECT_IS_TOP_LEVEL ON)
+	else ()
+		set(PROJECT_IS_TOP_LEVEL OFF)
+	endif ()
+endif ()
 
-add_library(GKlib ${GKlib_sources} ${win32_sources})
+# Specify C standard
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS ON)
 
-if(UNIX)
-  target_link_libraries(GKlib m)
-endif(UNIX)
+if (NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release)
+endif ()
 
-include_directories("test")
-add_subdirectory("test")
+#[==============================================================================================[
+#                                            Options                                            #
+]==============================================================================================]
 
-install(TARGETS GKlib
-  ARCHIVE DESTINATION lib/${LINSTALL_PATH}
-  LIBRARY DESTINATION lib/${LINSTALL_PATH})
-install(FILES ${GKlib_includes} DESTINATION include/${HINSTALL_PATH})
+include(CMakeDependentOption)
+include(FeatureSummary)
+option(GKLIB_TESTS "GKlib: Build unit tests" ${PROJECT_IS_TOP_LEVEL})
+option(GKLIB_SHARED_LIBS "GKlib: Build as a shared library" ${PROJECT_IS_TOP_LEVEL})
+option(GKLIB_INSTALL "GKlib: Install project" ${PROJECT_IS_TOP_LEVEL})
+
+# TODO: Discuss these options. Non-standard and easily conflicting
+option(GKLIB_ASSERT "GKlib: Turn asserts on" OFF)
+option(GKLIB_ASSERT2 "GKlib: Additional assertions" OFF)
+option(GKLIB_NO_X86 "GKlib: Enable NO_X86 support" ON)
+option(GKLIB_GPROF "GKlib: Add gprof support" OFF)
+
+option(GKLIB_OpenMP "GKlib: Enable OpenMP support" OFF)
+add_feature_info(GKLIB_OpenMP GKLIB_OpenMP "OpenMP support")
+option(GKLIB_PCRE "GKlib: Enable PCRE support" OFF)
+add_feature_info(GKLIB_PCRE GKLIB_PCRE "PCRE support")
+cmake_dependent_option(GKLIB_GKREGEX "GKlib: Enable GKREGEX support" OFF "NOT GKLIB_PCRE" OFF)
+add_feature_info(GKLIB_GKREGEX GKLIB_GKREGEX "GKREGEX support")
+option(GKLIB_GKRAND "GKlib: Enable GKRAND support" OFF)
+add_feature_info(GKLIB_GKRAND GKLIB_GKRAND "GKRAND support")
+
+#[==============================================================================================[
+#                                     Project configuration                                     #
+]==============================================================================================]
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+# Include basic tools
+include(FetchContent)
+if (GKLIB_INSTALL)
+	include(CMakePackageConfigHelpers)
+	if (UNIX)
```

Comment:
This should be unconditional, it's not an issue for this on Windows. And also, you use e.g. `CMAKE_INSTALL_BINDIR` unconditionally later on which expands to nothing on Windows currently.